lib/model/query/forms.js

```diff
@@ -613,9 +614,11 @@ where "acteeId"=${acteeId} and "deletedAt" is null`)
   .then(map(_unjoiner));
 
 // there are many combinations of required fields here so we compose our own extender variant.
-const _getSql = ((fields, extend, options, version, deleted = false, actorId) => sql`
+const _getSql = ((fields, extend, options, version, deleted = false, actorId) => {
+  const { enketoId, ...condition } = options.condition;
+  return sql`
 select ${fields} from forms
-left outer join form_defs on ${versionJoinCondition(version)}
+${enketoId ? sql`join` : sql`left outer join`} form_defs on ${versionJoinCondition(version, enketoId)}
```

Comment:
This is necessary to fetch the correct form_def. Provided `enketoId` can belong to the published Form or the draft one.

Another approach I was considering was to create a new function to first find out for which version provided `enketoId` belongs and then fetch that Form. If the current change is making already complicated query more complicated then I am open for creating separate query.